packages/backend/src/api/controllers/liveness/groupByType.ts

```diff
@@ -0,0 +1,33 @@
+import { LivenessRecordWithType } from '../../../peripherals/database/LivenessRepository'
+
+export interface GroupedByType {
+  batchSubmissions: {
+    records: LivenessRecordWithType[]
+  }
+  stateUpdates: {
+    records: LivenessRecordWithType[]
+  }
+  proofSubmissions: {
+    records: LivenessRecordWithType[]
+  }
+}
+
+export function groupByType(records: LivenessRecordWithType[]): GroupedByType {
```

Comment:
Instead of filtering 3 times, do one forEach loop and push to the according pre-created arrays